src/cargo/ops/cargo_compile.rs

```diff
@@ -361,6 +363,16 @@ fn generate_targets<'a>(pkg: &'a Package,
             Ok(targets)
         }
     };
+
+    targets.map(|x| x.iter().filter_map(|&(t, p)| {
```

Comment:
Hm perhaps this changed from before?  It looks like the `map` part of `filter_map` isn't being used, so this may be able to just be a `filter`.

It may also be more clear here to explicitly use if/else instead of a match for the logic, the multi-line guard for one of the arms was a little jarring.